### PR TITLE
bug fixes and testing with different LRs

### PR DESCRIPTION
**Pull request by sukruc**
fixed a problem which results in false gradient explosion alarm

### CODE REVIEW

File: Gradient Descent.ipynb [hunk 1]
```diff
@@ -47,25 +48,34 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def gradient_descent (X, y,tolerance=10**-6,  learning_rate=0.01, max_iter=10000, normalize=False):\n",
+    "def gradient_descent (X, y,tolerance=10**-6,  learning_rate=0.01, max_iter=10000, normalize=False, verbose=0,\n",
```
Comment (aatcili) [inline Gradient Descent.ipynb:51]: There are too much parameters

File: Gradient Descent.ipynb [hunk 2]
```diff
@@ -79,15 +89,115 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "a =  0.0616501782198029\n",
-      "Gradient =  -1.822597334277809e-06\n",
```
Comment (aatcili) [inline Gradient Descent.ipynb:83]: cluttering the notebook

File: Gradient Descent.ipynb [hunk 3]
```diff
@@ -166,38 +444,113 @@
...
...fY77j+OqO6inv9oHfG+Y2TeCY/6mmV0bT9UFcM4l8ovcydo/AWcD5cCrwAVx1zVArROBi4PbNcBbwAXAQ8B/jru+09T+DjD2hG3/BWgKbjcBD8dd52neJx+Qu9ij5I43cCVwMbDhdMcXuAH4V3LXklwKvFRidX8ayAS3H+5T99S++8X9NUDt/b43gn+nrwIVwLQgc9Jx/w75fCW5c+9d5sA5dxzoWeag5Djn2pxzLwe3DwIbyV3Jm1QLgObgdjNwa4y1nM484E/OuXfjLqQ/zrl/B/acsHmg47sAeMzlvAiMMrOJQ1PpR/VXt3NulXOuM7j7IrlrXErOAMd8IAuAJ51zx5xzbwNbyGVPyUtyuPe3zEHJB6aZTQUuAl4KNv1d8Gfso6U2vBFwwCozWxcsGQEwwTnXBrn/uIDxsVV3ercDP+tzv9SPNwx8fJP0nv88ub8yekwzs1fM7N/M7FNxFXUa/b03knTMPyLJ4Z7XMgelxMxGAE8DX3bOHQCWAecAFwJtwH+LsbyBXO6cuxi4HviimV0Zd0H5Ci6iuwV4KtiUhON9Kol4z5vZg0An8ESwqQ04yzl3EfBV4F/MrDau+gYw0HsjEce8P0kO90Qtc2BmZeSC/Qnn3DMAzrkdzrku51w38BNK8M8959z7wfedwLPkatzRMxwQfN8ZX4WndD3wsnNuByTjeAcGOr4l/543s0XATcDnXDBoHQxp7A5uryM3bn1efFWe7BTvjZI/5gNJcrgnZpkDy60v/FNgo3Pue3229x0v/Utgw4mPjZOZVZtZTc9tcifMNpA7zouC3RYBK+Kp8LQ+S58hmVI/3n0MdHxXAncFs2YuBfb3DN+UAst9SM/XgVucc0f6bB9nuc95wMzOBhqArfFU2b9TvDdWArebWYWZTSNX+9qhrq8gcZ/RDfNFbvbAW+Q6gQfjrucUdV5B7k+514D1wdcNwOPA68H2lcDEuGs9oe6zyc0UeBV4o+cYA2OANcDm4Htd3LX2U3sVsBsY2WdbyR1vcv/5tAEd5LrEuwc6vuSGCH4UvN9fBxpLrO4t5Mane97jPw72/avg/fMq8DJwcwke8wHfG8CDwTF/E7g+7vdMvl9afkBExENJHpYREZEBKNxFRDykcBcR8ZDCXUTEQwp3EREPKdxFRDykcBcR8dD/Bx9tvBYg3oOUAAAAAElFTkSuQmCC\n",
```
Comment (aatcili) [inline Gradient Descent.ipynb:516]: where are the blue and orange lines?

**Review by aatcili — APPROVED** [review]
